src/NServiceBus.RabbitMQ/Sending/MessageDispatcher.cs

```diff
@@ -21,14 +20,17 @@ public Task Dispatch(TransportOperations operations, ContextBag context)
 
             try
             {
-                var tasks = new List<Task>(operations.UnicastTransportOperations.Count() + operations.MulticastTransportOperations.Count());
+                var unicastTransportOperations = (List<UnicastTransportOperation>)operations.UnicastTransportOperations;
+                var multicastTransportOperations = (List<MulticastTransportOperation>)operations.MulticastTransportOperations;
```

Comment:
Yeah, I was thinking this one would be a bit controversial, for the exact reason you pointed out. I didn't just go for it since the goal here is removing allocations and calling `ToList` would allocate.

However, that allocation means core has changed, and an allocation vs. blowing up is definitely preferable. Gives us time to react to the core change without having to have them perfectly in sync.

Will make this change.